brave/brave-core
# AIChat: introducing slash tools

Resolves https://github.com/brave/brave-browser/issues/37294

This PR adds the ability to allow users to select actions which are ready-made prompts to the remote LLM.

The following UX is possible:

1. typing "/" will open the tools menu
2. you can filter an action when the tools menu open via "/"
3. you can press backspace on an empty input box to remove an action or press the "x"



https://github.com/brave/brave-core/assets/8665427/b36c7f93-13e7-42cf-9272-57aaf79b5cc0




## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
There are some UX nuances:

1. type “/” select an action the dropdown should close
5. type “/” scroll in the dropdown up and down and put your tap outside of the dropdown (it should close)
6. type "/" select an action and press "x" button
7. type "/" select an action with your keyboard
8. click "/" select an action
9. type "/" select an action and remove it. click on "/" select an action and remove it
10. while the action is selected pressing backspace on an empty input box should remove actions
11. type "/" and start filtering
